---
title: 'Mothur v.1.32.0'
---
We have released mothur v.1.32.0 and have a number of exciting new
features and updates to tell you about. The main new feature is to
incorporate the LEfSe software package from Curtis Huttenhower\'s lab
into mothur as the [lefse](lefse) command. Alternatively, you
could use the [make.lefse](make.lefse) command to generate a
file to use as input to LEfSe. They should produce the same output, let
us know if you think we\'re missing anything. We also incorporated the
Kruskal-Wallis non-parametric test (think ANOVA) as the
[kruskal.wallis](kruskal.wallis) command - this is part of
the LEfSe algorithm. Among the new features we\'ve renamed a couple of
commands (see below) and improved the output generated by the
[classify.rf](classify.rf) command (formerly the
[classify.shared](classify.shared) command. We are putting
the finishing touches on [classify.svm](classify.svm), which
will be similar to [classify.rf](classify.rf) (random
forest), but uses support vector machines (SVMs). As always, there have
been a number of bugs reported that we\'ve addressed in this release. We
also noticed that the greengenes folks expanded their database and
taxonomy and we have updated the mothur-compatible versions of those.
Going forward we\'re looking to add a number of new commands for the
next release including a SRA submission tool. If you have any
suggestions for other commands or options, let us know!

Finally, there are still seats available for the R workshop in November
and the mothur workshop in December. Consider coming to one or both -
Detroit is beautiful that time of year.

## New commands

-   [rename.seqs](rename.seqs) -
    [https://forum.mothur.org/viewtopic.php?f=5&t=1918](https://forum.mothur.org/viewtopic.php?f=5&t=1918)
-   [make.lefse](make.lefse) - creates a
    [LEfSe](https://huttenhower.sph.harvard.edu/galaxy/root?tool_id=lefse_upload)
    compatible input file from mothur\'s output files.
-   [kruskal.wallis](kruskal.wallis) - find the Kruskal
    Wallis one way analysis of variance for each OTU.
-   [lefse](lefse) - modelled after the
    [LEfSe](https://huttenhower.sph.harvard.edu/galaxy/root?tool_id=lefse_upload)
    program written by the Huttenhower lab.

## Feature updates

-   [get.metacommunity](get.metacommunity) changed to
    [get.communitytype](get.communitytype).
-   [classify.shared](classify.shared) changed to
    [classify.rf](classify.rf)
-   [classify.seqs](classify.seqs) - removed extra name
    checks for speed up of reading taxonomy file. -
    [https://forum.mothur.org/viewtopic.php?f=4&p=6608&sid=42525a60989d015feb8c10b2b5342691](https://forum.mothur.org/viewtopic.php?f=4&p=6608&sid=42525a60989d015feb8c10b2b5342691)
-   [screen.seqs](screen.seqs) - allowed for fractional
    percent in the criteria parameter (e.g. criteria=97.5)
-   [classify.rf](classify.rf) - improved output
-   [make.contigs](make.contigs) - added findex and rindex
    parameters

## Bug fixes

-   [unique.seqs](unique.seqs) - segfault with count file
    containing group info. - fixed 1.31.1
-   [get.seqs](get.seqs) - dups=f, renaming issue,
    [https://forum.mothur.org/viewtopic.php?f=3&t=2371](https://forum.mothur.org/viewtopic.php?f=3&t=2371). - fixed
    1.31.1
-   [make.contigs](make.contigs) -
    [https://forum.mothur.org/viewtopic.php?f=3&t=2451](https://forum.mothur.org/viewtopic.php?f=3&t=2451) - fixed
    1.31.2
-   [sffinfo](sffinfo) - when parsing an sff file using an
    oligos file mothur did not adjust the index offset and index length
    values in the common header.
-   [get.communitytype](get.communitytype) order of sample
    names was not correct in .mix.posterior file if the shared file was
    not sorted.
-   [metastats](metastats) - generate p values for sparse
    data using fisher\'s exact test instead of permutted values
-   [split.groups](split.groups) - not splitting properly
    with a count file.
-   [make.contigs](make.contigs) - if file option is used
    with group provided, and one or more files contain less good reads
    than number of processors, group assignments were incorrect. -
    [https://forum.mothur.org/viewtopic.php?f=4&t=2571&p=7025#p7025](https://forum.mothur.org/viewtopic.php?f=4&t=2571&p=7025#p7025)
-   [trim.seqs](trim.seqs) - if primer length + barcode
    length + pdiffs + bdiffs \> sequence length, mothur crashed.
    Sequence should be scrapped.
-   [shhh.flows](shhh.flows) - sequences longer than 1024
    caused crash on linux.
-   [make.biom](make.biom) - segfault with multiple labels in
    shared file. -
    [https://forum.mothur.org/viewtopic.php?f=4&t=2585&p=7072#p7072](https://forum.mothur.org/viewtopic.php?f=4&t=2585&p=7072#p7072)
-   [screen.seqs](screen.seqs) - for count tables without
    group information, mothur was not creating the \*good.count\_table
    correctly. Which could cause fasta/count file mismatch errors
    downstream.
-   removed -m64 flag from makefile for linux
-   moved lookup file default position to
    pathToMothursExe/lookupFiles/LookUp\_Titanium.pat -
    [https://forum.mothur.org/viewtopic.php?f=4&t=1366&sid=7f52c79920891f1532992c30f8fe6a3a&start=10](https://forum.mothur.org/viewtopic.php?f=4&t=1366&sid=7f52c79920891f1532992c30f8fe6a3a&start=10)

## Changes to wiki

-   Added ITS database from Segre Lab
-   Updated the greengenes-formatted databases

## GUI

-   [https://forum.mothur.org/viewtopic.php?f=4&t=2508&p=7147#p7147](https://forum.mothur.org/viewtopic.php?f=4&t=2508&p=7147#p7147) -
    Preferences file menu option not working. Preferences file
    overwriting references.
-   when set.dir command was used logfile path name caused \"Could not
    open\" error.

## Registered users

2626
