---
title: 'mothur manual'
redirect_from: '/wiki/Mothur_manual'
---
The goal of mothur is to have a single resource to analyze molecular
data that is used by microbial ecologists. Many of these tools are
available elsewhere as individual programs and as scripts, which tend to
be slow or as web utilities, which limit your ability to analyze your
data. mothur offers the ability to go from raw sequences to the
generation of visualization tools to describe α and β diversity.
Examples of each command are provided within their specific pages, but
several users have provided several [analysis
examples](/wiki/analysis_examples), which use these commands. An
exhaustive list of the commands found in mothur is available within the
[ commands category index](/wiki/tags#commands). If you have
any questions, complaints, or praise, please do not hesitate to [email Pat](mailto:pschloss@umich.edu) or to use the [user forum](https://forum.mothur.org):

-   [general operations](/wiki/General_operations)
-   [sequence processing](/wiki/Sequence_processing)
-   [chimera detection commands](/wiki/Chimera_Detection_Commands)
-   [otu-based approaches](/wiki/OTU-based_approaches)
-   [hypothesis testing approaches](/wiki/Hypothesis_testing_approaches)
-   [frequently asked questions](/wiki/Frequently_asked_questions)
-   [phylotype analysis](/wiki/Phylotype_analysis)
-   [file types](/wiki/tags#file_types)

Remember to cite mothur:

> Schloss, P.D., et al., Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41

A BibTeX entry for LaTeX users:

```TeX
@article {Schloss7537,
	author = {Schloss, Patrick D. and Westcott, Sarah L. and Ryabin, Thomas and Hall, Justine R. and Hartmann, Martin and Hollister, Emily B. and Lesniewski, Ryan A. and Oakley, Brian B. and Parks, Donovan H. and Robinson, Courtney J. and Sahl, Jason W. and Stres, Blaz and Thallinger, Gerhard G. and Van Horn, David J. and Weber, Carolyn F.},
	title = {Introducing mothur: Open-Source, Platform-Independent, Community-Supported Software for Describing and Comparing Microbial Communities},
	volume = {75},
	number = {23},
	pages = {7537--7541},
	year = {2009},
	doi = {10.1128/AEM.01541-09},
	publisher = {American Society for Microbiology Journals},
	abstract = {mothur aims to be a comprehensive software package that allows users to use a single piece of software to analyze community sequence data. It builds upon previous tools to provide a flexible and powerful software package for analyzing sequencing data. As a case study, we used mothur to trim, screen, and align sequences; calculate distances; assign sequences to operational taxonomic units; and describe the α and β diversity of eight marine samples previously characterized by pyrosequencing of 16S rRNA gene fragments. This analysis of more than 222,000 sequences was completed in less than 2 h with a laptop computer.},
	issn = {0099-2240},
	URL = {https://aem.asm.org/content/75/23/7537},
	eprint = {https://aem.asm.org/content/75/23/7537.full.pdf},
	journal = {Applied and Environmental Microbiology}
}
```